classes/Cart.php

```diff
@@ -2044,7 +2047,16 @@ public function getOrderTotal(
         // If no specific products list is provided, we get the full list of products in cart
         // In most cases, we calculate the total for all products in cart
         if (null === $products) {
-            $products = $this->getProducts(false, false, null, true, $keepOrderPrices);
+            if ($type == Cart::ONLY_PRODUCTS_WITHOUT_GIFTS) {
```

Comment:
This is the main change in the PR. If we want a total without gifts, let's return it immediately. Not subtract the gifts from all products. :-)